### ISSUE TITLE
Add Feature: Gravatar automatically pulled to profile avatar

### ISSUE THREAD

**Issue by jLouzado** [wekan/wekan#824]
Right now you have to manually add an avatar when added to a board, this could be simplified by just querying for a gravatar image (if available) or defaulting to the use of Initials.

**Comment by lifehome** [comment]
Hello there, is there any update on this? It seems the docker image does not contain such feature, I wonder if the master head does?

**Comment by xet7** [comment]
@lifehome 

No, this does not exists yet.

**Comment by xet7** [comment]
Using Gravatar is not compliant to EU GDPR regulation.

**Comment by zewa666** [comment]
Any updates on the GDPR status of Gravatar @xet7 ?

**Comment by xet7** [comment]
@zewa666 

Yes. Using Gravatar is not compliant with GDPR. Gravatar support will not be implemented in Wekan.

Similar issue here:
https://lists.mailman3.org/archives/list/mailman-users@mailman3.org/thread/H5I7MO7Z5FN3LLEVGU4EUW6K7FNWJ2DT/

**Comment by zewa666** [comment]
thanks for the link 👍

**Comment by lifehome** [comment]
May I know why opt-in for Gravatar is not even considered? I am surely aware the GDPR but not everyone lives in the EU, nor all instances being regulated by the laws of EU.

**Comment by xet7** [comment]
@lifehome 

GDPR affects also all other companies that do business with EU, like companies at USA. If at some other country there is not such strong privacy protections, most likely govenment can hijack those connections to gravatar, potentially inserting something malicious. So having Gravatar support is security bug.

Gravatar does not work in private intranets where is no direct internet connection, or where is some kind of proxy.  For those environments, Gravatar would not work, and even possibility of that option would make usage non-GDPR compliant. Same is with Sandstorm, Wekan is inside secure sandbox, with no access from Sandstorm sandbox to outside to Internet, if no Sandstorm-compatible way is implemented.

Wekan does not load any external resources from internet, like fonts, CSS, etc. That means Wekan is currently safe to use in company intranet use, in healthcare, etc. It's a strong guarantee, I do not want to break that.

**Comment by xet7** [comment]
@lifehome 

I would only accept General Avatar feature, with settings configured in environment variables, 
like URL that can be at gravatar or local intranet server, like http://192.168.1.200/avatars/username.png, with username text replaced with Wekan username.

There should be no UI in Admin Panel, all should be configured in environment variables of standalone/docker/snap version.

This would be coded and submitted as pull request to Wekan devel branch by somebody that really wants this feature.

For implementing this, see [Developer Documentation](https://github.com/wekan/wekan/wiki/Developer-Documentation) and [Excellent example how pull requests are improved and integrated, and not needed commits removed](https://github.com/wekan/wekan/pull/1470)

**Comment by xi67** [comment]
There is no private data transmitted to Gravatar.
The service use a hash (MD5 of email address (not bijective so you can't retrieve email adress from the hash)) : https://fr.gravatar.com/site/implement/profiles/
So Wekan does not transmit any private data in requesting if there is an avatar for this hash.

In this logic, Wekan is not responsible for data present in Gravatar Website, it just request it.
I'm not an lawer but i don't see the problem around GDPR because the only data transmitted doas not contain any private information.
